x-pack/plugins/security/server/authentication/authenticator.test.ts

```diff
@@ -1318,6 +1316,59 @@ describe('Authenticator', () => {
       expectAuditEvents({ action: 'user_login', outcome: 'failure' });
     });
 
+    // TODO: Is it realistic that all providers choose not to handle it?
+    it('fails if session is unhandled by all providers and getting the current session results in an SessionExpiredError for redirectable requests.', async () => {
```

Comment:
This comment applies for this whole block of new tests:

If I understand the flow correctly, providers won't have an opportunity to attempt to handle the request if the session is expired/invalid. Expired/invalid sessions are handled before we enter the provider loop to attempt provider-specific authentication. I think the tests you have here are still valid, but we likely don't need to say that the `session is unhandled by all providers`